bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -363,11 +371,14 @@ public void dispose() {
             this.metadataRefreshJob = null;
         }
 
-        ServiceRegistration<?> localServiceRegistration = serviceRegistration;
-        if (localServiceRegistration != null) {
-            // remove trustmanager service
-            localServiceRegistration.unregister();
-            serviceRegistration = null;
+        try {
+            HttpClient localHttpClient = this.httpClient;
+            if (localHttpClient != null) {
+                localHttpClient.stop();
+            }
+            localHttpClient = null;
```

Comment:
```suggestion
            this.httpClient = null;
```

Reply:
I see `localHttpClient` is still being nulled, I don't think there is any reason for that? I assumed the local variable was used by mistake rather than the instance variable.

Reply:
fixed